packages/lib-user/dev/components/App/App.js

```diff
@@ -0,0 +1,134 @@
+
+import { usePanoptesUser } from '@hooks/index.js'
+import { GroupStats, UserStats } from '@components/index.js'
+
+function App ({
+  groups = null,
+  users = null
+}) {
+  const [loading, setLoading] = useState(false)
+  const [userAuth, setUserAuth] = useState(null)
+
+  const { data: user, error, isLoading: userLoading } = usePanoptesUser(oauth)
+
+  useEffect(() => {
+    async function initAuthorization () {
+      setLoading(true)
+
+      try {
+        const userAuth = await oauth.init('357ac7e0e17f6d9b05587477ca98fdb69d70181e674be8e20142e1df97a84d2d');
+        setUserAuth(userAuth);
+        setLoading(false);
+      } catch (error) {
+        console.error(error);
+        setLoading(false);
+      }
+    }
+
+    initAuthorization()
+  }, [])
+
+  const login = () => oauth.signIn(window.location.origin)
+  const logout = () => oauth.signOut().then(setUserAuth)
+
+  let content = (
+    <div>
+      <h2>Key Components - urls (zooniverse.org/...)</h2>
+      <ul>
+        <li>homepage</li>
+        <li>
+          profile page (public) - users/[login]
+          <ul>
+            <li>
+              <a href={`./?users=${user?.login ?? "[login]"}/stats`}>user stats page (private) - users/{user?.login ?? "[login]"}/stats</a>
+              <ul>
+                <li>certificate - users/{user?.login ?? "[login]"}/stats/certificate</li>
+              </ul>
+            </li>
+          </ul>
+        </li>
+        <li>
+          groups list - TBD
+          <ul>
+            <li>
+              <a href="./?groups=[user_group_id]">group stats page - groups/[id]</a>
+              <ul>
+                <li>contributors - groups/[id]/contributors</li>
+              </ul>
+            </li>
+          </ul>
+        </li>
+      </ul>
+    </div>
+  )
+
+  if (groups) {
+    const subpaths = groups.split('/')
+  
+    if (subpaths[0] === '[user_group_id]') {
+      content = <p>In the url query param <code>?groups=</code>, please replace <code>[user_group_id]</code> with a user group id.</p>
+    } else if (subpaths[1] === 'contributors') {
+      content = <p>Group contributors component goes here.</p>
+    } else {
+      const groupID = subpaths[0] || ''
+
+      content = (
+        <GroupStats
+          authClient={oauth}
+          groupID={groupID}
+        />
+      )
+    }
+  }
+
+  if (users) {
+    const subpaths = users.split('/')
+
+    if (subpaths[0] === '[login]') {
+      content = <p>In the url query param <code>?users=</code>, please replace <code>[login]</code> with a user login.</p>
+    } else if (subpaths[1] === 'stats') {
+      const login = subpaths[0] || ''
+
+      content = (
+        <UserStats
+          authClient={oauth}
+          login={login}
+        />
+      )
+    }
+  }
+
+  return (
+    <main>
+      <header>
+        <h1><a href="./">lib-user</a></h1>
+        {userAuth ? (
+          <button onClick={logout}>
+            Logout
+          </button>
```

Comment:
When I click this button, I get an unhandled error `Request has been terminated` in the browser. Is that intentionally left unhandled in this PR? Totally fine if not, but wanted to check that you're aware.

Reply:
I was able to recreate this issue, and had a different issue when first starting the dev server/App.js, but am now unable to reliably reproduce. I don't think the dev server/App.js is bug free, but I think I'd like to punt on further refactoring for the moment and can revisit in a subsequent PR?

Reply:
Sounds good!